src/ekf.cpp

```diff
@@ -362,7 +362,26 @@ namespace RobotLocalization
     estimateErrorCovariance_ = (transferFunctionJacobian_ *
                                 estimateErrorCovariance_ *
                                 transferFunctionJacobian_.transpose());
-    estimateErrorCovariance_.noalias() += (processNoiseCovariance_ * delta);
+
+    // Compute processNoiseCovariance scaling if nessesary
+    Eigen::MatrixXd processNoiseCovariance = processNoiseCovariance_;
```

Comment:
OK, I see... we need this on the `predict` because the velocity state changes, as well as the orientation, and it's needed to scale the process noise. Good.